framework_tvb/tvb/core/services/simulator_service.py

```diff
@@ -226,13 +237,200 @@ def async_launch_and_prepare_pse(self, burst_config, user, project, simulator_al
+                                            "a connectivity!")
+        return form
+
+    @staticmethod
+    def _compute_conn_branch_conditions(is_branch, simulator):
+        if is_branch:
+            conn = dao.get_datatype_by_gid(simulator.connectivity.hex)
+            if conn.number_of_regions:
+                return FilterChain(fields=[FilterChain.datatype + '.number_of_regions'],
+                                   operations=["=="], values=[conn.number_of_regions])
+        return None
+
+    def prepare_next_fragment_after_surface(self, simulator, rendering_rules, project_id):
+        if simulator.surface:
+            return self.prepare_cortex_fragment(simulator, rendering_rules, SimulatorWizzardURLs.SET_CORTEX_URL,
+                                                project_id)
+        return self.prepare_stimulus_fragment(simulator, rendering_rules, False, SimulatorWizzardURLs.SET_STIMULUS_URL,
+                                              project_id)
+
+    def prepare_cortex_fragment(self, simulator, rendering_rules, form_action_url, project_id):
+        surface_index = load_entity_by_gid(simulator.surface.surface_gid)
+        form = SimulatorRMFragment(surface_index, simulator.connectivity)
+        rm_fragment = self.algorithm_service.prepare_adapter_form(form_instance=form, project_id=project_id)
+        rm_fragment.fill_from_trait(simulator.surface)
+
+        rendering_rules.form = rm_fragment
+        rendering_rules.form_action_url = form_action_url
+        return rendering_rules.to_dict()
+
+    def prepare_stimulus_fragment(self, simulator, rendering_rules, is_surface_simulation, form_action_url, project_id):
+        form = SimulatorStimulusFragment(is_surface_simulation)
+        stimuli_fragment = self.algorithm_service.prepare_adapter_form(form_instance=form, project_id=project_id)
+        stimuli_fragment.fill_from_trait(simulator)
+
+        rendering_rules.form = stimuli_fragment
+        rendering_rules.form_action_url = form_action_url
+        return rendering_rules.to_dict()
+
+    def prepare_next_fragment_after_noise(self, simulator, is_branch, rendering_rules):
+        if isinstance(simulator.integrator.noise, AdditiveNoiseViewModel):
+            rendering_rules.is_branch = is_branch
+            return self.prepare_monitor_fragment(simulator, rendering_rules, SimulatorWizzardURLs.SET_MONITORS_URL)
+
+        equation_form = get_form_for_equation(type(simulator.integrator.noise.b))()
+        equation_form.equation.data = simulator.integrator.noise.b.equation
+        equation_form.fill_from_trait(simulator.integrator.noise.b)
+
+        rendering_rules.form = equation_form
+        rendering_rules.form_action_url = SimulatorWizzardURLs.SET_NOISE_EQUATION_PARAMS_URL
+        return rendering_rules.to_dict()
+
+    @staticmethod
+    def prepare_monitor_fragment(simulator, rendering_rules, form_action_url):
+        monitor_fragment = SimulatorMonitorFragment(simulator.is_surface_simulation)
```

Comment:
I see "fragments" related code moved here. It seems a bit counter intuitive, as it is GUI related code, which should not really be in the service layer. I suggest code near the simulator fragments classes, near the simulator form classes or in the controller